src/components/binding/binding-panel.js

```diff
@@ -213,13 +187,44 @@ genotet.BindingPanel.prototype.updateTracks = function() {
       this.container.find('#genes #track-' + index).remove();
     }
   }
+
+  this.signal('loadBindingList');
+
+  this.container.find('#genes .glyphicon-remove')
+    .css('display', this.data.tracks.length == 1 ? 'none' : '');
+};
+
+/**
+ * Updates the track list for panel.
+ */
+genotet.BindingPanel.prototype.updateTrackList = function() {
+  var fileNames = genotet.data.bindingFiles.map(function(dataInfo) {
+    return {
+      id: dataInfo.fileName,
+      text: dataInfo.gene + ' (' + dataInfo.fileName + ')'
+    };
+  });
   this.data.tracks.forEach(function(track, index) {
     var ui = this.container.find('#genes #track-' + index);
+
     if (!ui.length) {
```

Comment:
A bit more on this: I think you have two options:
- Create a single updateTracks function that handles addition/removal/update in a single entry. So for all track edit events you just call updateTracks. updateTracks determines when to add/remove DOMs when necessary.
- Have three handlers for addition/removal/update respectively so that each handles its own job. If so, then 'addTrack' event should only trigger addTrack handler. 'updateTracks' event should only trigger updateTrack handlers, in which no track addition/removal are involved.